examples/gno.land/p/demo/json/decode_test.gno

```diff
@@ -0,0 +1,567 @@
+        {"id": "CopySVG", "label": "Copy SVG"},
+        {"id": "ViewSVG", "label": "View SVG"},
+        {"id": "ViewSource", "label": "View Source"},
+        {"id": "SaveAs", "label": "Save As"},
+        null,
+        {"id": "Help"},
+        {"id": "About", "label": "About Adobe CVG Viewer..."}
+    ]
+}}`
+
+var bookStore = `{ "store": {
+    "book": [ 
+      { "category": "reference",
+        "author": "Nigel Rees",
+        "title": "Sayings of the Century",
+        "price": 8.95
+      },
+      { "category": "fiction",
+        "author": "Evelyn Waugh",
+        "title": "Sword of Honour",
+        "price": 12.99
+      },
+      { "category": "fiction",
+        "author": "Herman Melville",
+        "title": "Moby Dick",
+        "isbn": "0-553-21311-3",
+        "price": 8.99
+      },
+      { "category": "fiction",
+        "author": "J. R. R. Tolkien",
+        "title": "The Lord of the Rings",
+        "isbn": "0-395-19395-8",
+        "price": 22.99
+      }
+    ],
+    "bicycle": {
+      "color": "red",
+      "price": 19.95
+    }
+  }
+}`
+
+// Examples from https://json.org/example.html
+func TestUnmarshal(t *testing.T) {
+	tests := []struct {
+		name  string
+		value string
+	}{
+		{
+			name:  "glossary",
+			value: glossary,
+		},
+		{
+			name:  "menu",
+			value: file,
+		},
+		{
+			name:  "widget",
+			value: widget,
+		},
+		{
+			name:  "web-app",
+			value: webApp,
+		},
+		{
+			name:  "SVG Viewer",
+			value: svgViewer,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := Unmarshal([]byte(test.value))
+			if err != nil {
+				t.Errorf("Error on Unmarshal: %s", err.Error())
+			}
+		})
+	}
+}
+
+func TestUnmarshalSafe(t *testing.T) {
+	json := []byte(bookStore)
+	safe, err := UnmarshalSafe(json)
+	if err != nil {
+		t.Errorf("Error on Unmarshal: %s", err.Error())
+	} else if safe == nil {
+		t.Errorf("Error on Unmarshal: safe is nil")
+	} else {
+		root, err := Unmarshal(json)
+		if err != nil {
+			t.Errorf("Error on Unmarshal: %s", err.Error())
+		} else if root == nil {
+			t.Errorf("Error on Unmarshal: root is nil")
+		} else if !bytes.Equal(root.source(), safe.source()) {
+			t.Errorf("Error on UnmarshalSafe: values not same")
+		}
+	}
+}
+
+// BenchmarkGoStdUnmarshal-8   	   61698	     19350 ns/op	     288 B/op	       6 allocs/op
+// BenchmarkUnmarshal-8        	   45620	     26165 ns/op	   21889 B/op	     367 allocs/op
+//
+// type bench struct {
+// 	Name  string `json:"name"`
+// 	Value int    `json:"value"`
+// }
+
+// func BenchmarkGoStdUnmarshal(b *testing.B) {
+// 	data := []byte(webApp)
+// 	for i := 0; i < b.N; i++ {
+// 		err := json.Unmarshal(data, &bench{})
+// 		if err != nil {
+// 			b.Fatal(err)
+// 		}
+// 	}
+// }
+
+// func BenchmarkUnmarshal(b *testing.B) {
+// 	data := []byte(webApp)
+// 	for i := 0; i < b.N; i++ {
+// 		_, err := Unmarshal(data)
+// 		if err != nil {
+// 			b.Fatal(err)
+// 		}
+// 	}
+// }
```

Comment:
I would uncomment these, even if we have no exact way of running them natively in Gno